portal-api/src/v1/roles/validate-user-is-acting-on-self-to-change-password-or-is-admin.js

```diff
@@ -0,0 +1,22 @@
+const { HttpStatusCode } = require('axios');
+const CONSTANT = require('../../constants');
+/**
+ * Creates a middleware that returns a 401 error if the current user logs as admin or other user, and calls the next handler otherwise.
+ * @param {{ allowedRoles: string[] }}
+ * @returns {(req, res, next) => void}
+ */
+const validateUserIsActingOnSelfToChangePasswordOrIsAdmin = (req, res, next) => {
+  if (!req.user) {
+    return res.status(HttpStatusCode.Unauthorized).send('Unauthorized');
+  }
+
+  if (req?.user?._id !== req?.params?._id && !req?.user?.roles?.includes(CONSTANT.ADMIN) && !req?.user?.roles?.includes(CONSTANT.MAKER)) {
```

Comment:
Simplify the compound condition.

Reply:
Will this prevent current logged in `maker` and `admin` from changing their own password?

Reply:
Simplified the compound condition. Yes, `maker` and `admin` can change their own password.